--- a/feeds/luci/applications/luci-app-ddns/root/usr/share/rpcd/ucode/ddns.uc
+++ b/feeds/luci/applications/luci-app-ddns/root/usr/share/rpcd/ucode/ddns.uc
@@ -224,42 +224,62 @@
 			let res = {};
 			let cache = {};
 
-			const hasCommand = (command) => { return (system(`command -v ${command} 1>/dev/null`) == 0) ? true : false };
+			function hasCommand(command) {
+				return (system(`command -v ${command} 1>/dev/null`) == 0) ? true : false;
+			};
 
-			const hasWget = () => {
-				return cache.has_wget ??= hasCommand('wget');
+			function hasWget() {
+				if (cache.has_wget == null)
+					cache.has_wget = hasCommand('wget');
+				return cache.has_wget;
 			}
 
-			const hasWgetSsl = () => {
-				return cache.has_wgetssl ??= hasWget() && system(`wget 2>&1 | grep -iqF 'https'`) == 0 ? true : false;
+			function hasWgetSsl() {
+				if (cache.has_wgetssl == null)
+					cache.has_wgetssl = hasWget() && system(`wget 2>&1 | grep -iqF 'https'`) == 0 ? true : false;
+				return cache.has_wgetssl;
 			};
 
-			const hasGNUWgetSsl = () => {
-				return cache.has_gnuwgetssl ??= hasWget() && system(`wget -V 2>&1 | grep -iqF '+https'`) == 0 ? true : false;
+			function hasGNUWgetSsl() {
+				if (cache.has_gnuwgetssl == null)
+					cache.has_gnuwgetssl = hasWget() && system(`wget -V 2>&1 | grep -iqF '+https'`) == 0 ? true : false;
+				return cache.has_gnuwgetssl;
 			};
 
-			const hasCurl = () => {
-				return cache.has_curl ??= hasCommand('curl');
+			function hasCurl() {
+				if (cache.has_curl == null)
+					cache.has_curl = hasCommand('curl');
+				return cache.has_curl;
 			};
 
-			const hasCurlSsl = () => {
-				return cache.has_curl_ssl ??= system(`curl -V 2>&1 | grep -qF 'https'`) == 0 ? true : false;
+			function hasCurlSsl() {
+				if (cache.has_curl_ssl == null)
+					cache.has_curl_ssl = system(`curl -V 2>&1 | grep -qF 'https'`) == 0 ? true : false;
+				return cache.has_curl_ssl;
 			};
 
-			const hasFetch = () => {
-				return cache.has_fetch ??= hasCommand('uclient-fetch');
+			function hasFetch() {
+				if (cache.has_fetch == null)
+					cache.has_fetch = hasCommand('uclient-fetch');
+				return cache.has_fetch;
 			};
 
-			const hasFetchSsl = () => {
-				return cache.has_fetch_ssl ??= stat('/lib/libustream-ssl.so') ? true : false;
+			function hasFetchSsl() {
+				if (cache.has_fetch_ssl == null)
+					cache.has_fetch_ssl = stat('/lib/libustream-ssl.so') ? true : false;
+				return cache.has_fetch_ssl;
 			};
 
-			const hasCurlPxy = () => {
-				return cache.has_curl_proxy ??= system(`grep -i 'all_proxy' /usr/lib/libcurl.so*`) == 0 ? true : false;
+			function hasCurlPxy() {
+				if (cache.has_curl_proxy == null)
+					cache.has_curl_proxy = system(`grep -i 'all_proxy' /usr/lib/libcurl.so*`) == 0 ? true : false;
+				return cache.has_curl_proxy;
 			};
 
-			const hasBbwget = () => {
-				return cache.has_bbwget ??= system(`wget -V 2>&1 | grep -iqF 'busybox'`) == 0 ? true : false;
+			function hasBbwget() {
+				if (cache.has_bbwget == null)
+					cache.has_bbwget = system(`wget -V 2>&1 | grep -iqF 'busybox'`) == 0 ? true : false;
+				return cache.has_bbwget;
 			};
 
 
@@ -271,7 +291,7 @@
 			res['has_forceip'] = hasGNUWgetSsl() || hasWgetSsl() || hasCurl() || hasFetch();
 			res['has_bindnet'] = hasCurl() || hasGNUWgetSsl();
 
-			const hasBindHost = () => {
+			function hasBindHost() {
 				if (cache['has_bindhost']) return cache['has_bindhost'];
 				const commands = ['host', 'khost', 'drill'];
 				for (let command in commands) {
@@ -287,17 +307,17 @@
 
 			res['has_bindhost'] = cache['has_bindhost'] || hasBindHost();
 
-			const hasHostIp = () => {
+			function hasHostIp() {
 				return hasCommand('hostip');
 			};
 
-			const hasNslookup = () => {
+			function hasNslookup() {
 				return hasCommand('nslookup');
 			};
 
 			res['has_dnsserver'] = cache['has_bindhost'] || hasNslookup() || hasHostIp() || hasBindHost();
 
-			const checkCerts = () => {
+			function checkCerts() {
 				let present = false;
 				for (let cert in glob('/etc/ssl/certs/*.crt', '/etc/ssl/certs/*.pem')) {
 					if (cert != null)
