--- a/feeds/luci/applications/luci-app-openclash/root/etc/init.d/openclash
+++ b/feeds/luci/applications/luci-app-openclash/root/etc/init.d/openclash
@@ -288,7 +288,7 @@
          done
       fi
       if [ -n "$wan6_dns" ]; then
-         echo "# Interface LAN6" > "$resolv_file"
+         echo "# Interface LAN6" >> "$resolv_file"
          for dns6 in $wan6_dns; do
             echo "nameserver $dns6" >> "$resolv_file"
          done
